index.js

```diff
@@ -1,654 +1,20 @@
-        object: 'arguments',
-        property: 'callee',
-        message: 'arguments.callee is deprecated',
-      },
-      {
-        object: 'global',
-        property: 'isFinite',
-        message: 'Please use Number.isFinite instead',
-      },
-      {
-        object: 'self',
-        property: 'isFinite',
-        message: 'Please use Number.isFinite instead',
-      },
-      {
-        object: 'window',
-        property: 'isFinite',
-        message: 'Please use Number.isFinite instead',
-      },
-      {
-        object: 'global',
-        property: 'isNaN',
-        message: 'Please use Number.isNaN instead',
-      },
-      {
-        object: 'self',
-        property: 'isNaN',
-        message: 'Please use Number.isNaN instead',
-      },
-      {
-        object: 'window',
-        property: 'isNaN',
-        message: 'Please use Number.isNaN instead',
-      },
-      {
-        property: '__defineGetter__',
-        message: 'Please use Object.defineProperty instead.',
-      },
-      {
-        property: '__defineSetter__',
-        message: 'Please use Object.defineProperty instead.',
-      },
-      {
-        object: 'Math',
-        property: 'pow',
-        message: 'Use the exponentiation operator (**) instead.',
-      },
-    ],
-    'no-return-assign': ['error', 'always'],
-    'no-return-await': ['error'],
-    'no-script-url': ['error'],
-    'no-self-compare': ['error'],
-    'no-sequences': ['error'],
-    'no-throw-literal': ['error'],
-    'no-unused-expressions': [
-      'error',
-      {
-        allowShortCircuit: false,
-        allowTernary: false,
-        allowTaggedTemplates: false,
-        enforceForJSX: false,
-      },
-    ],
-    'no-useless-concat': ['error'],
-    'no-useless-return': ['error'],
-    'no-void': ['error'],
-    'prefer-promise-reject-errors': ['error', {allowEmptyReject: true}],
-    'prefer-regex-literals': ['error', {disallowRedundantWrapping: true}],
-    radix: ['error'],
-    'vars-on-top': ['error'],
-    yoda: ['error'],
-  },
-
-  settings: {
-    // do not attempt to parse npm modules or non-JS files for exports
-    'import/ignore': ['node_modules', '.(png|svg|jpg|css|pdf)$'],
-
-    'import/extensions': ['.ts', '.cts', '.mts', '.tsx', '.js', '.jsx', '.mjs', '.cjs'],
-    'import/external-module-folders': ['node_modules', 'node_modules/@types'],
-    'import/parsers': {
-      '@typescript-eslint/parser': ['.ts', '.cts', '.mts', '.tsx'],
+import js from '@eslint/js'
+import ts from 'typescript-eslint'
+import prettier from 'eslint-config-prettier'
+import reactHooks from 'eslint-plugin-react-hooks'
+
+export default ts.config(
+  js.configs.recommended,
+  ...ts.configs.recommended,
+  {
+    plugins: {
+      'react-hooks': reactHooks,
     },
-    'import/resolver': {
-      node: {
-        extensions: ['.ts', '.cts', '.mts', '.tsx', '.js', '.jsx'],
-      },
+    rules: {
+      '@typescript-eslint/no-unused-vars': ['error', {ignoreRestSiblings: true}],
```

Comment:
Are there specific use-cases you saw either in existing codebases or generally that nudged you in the direction of changing the default of `ignoreRestSiblings`? It seems like a reasonable override but I can also see how it might not be necessary to override at all (and one less deviation from the baseline).

Reply:
Yeah, we use this pattern occasionally in internal codebases to omit properties from an object, e.g.:

```tsx
const {unused, ...rest} = props
return <button {...rest} />
```

Without this override, I don't see an ergonomic way to do this short of introducing an `omit` util. Do you?

Reply:
Talked offline a bit and it seems like since this is a pattern used enough internally and without an ergonomic alternative it justifies the override.

Thanks for providing the context @billyjanitsch !